### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by Mayureshd-18**
Fixed some typos

### CODE REVIEW

**Comment by Mayureshd-18** [comment]
@scottamain Please review these.
Regards

File: proposals/lifetimes-and-provenance.md [hunk 1]
```diff
@@ -220,7 +220,7 @@ NOTE: Internally to the compiler, references (which are always pointer sized)
 are treated as a register-passable types.  This means they compose correctly
 with implicit borrow semantics, and you can even pass a reference `inout` if you
 want to.  Such a thing is a "mutable reference to a reference", allowing the
-callee to mutate the callers reference.
+called to mutate the callers reference.
```
Comment (scottamain) [inline proposals/lifetimes-and-provenance.md:223]: "callee" is correct. I suppose "called" also works if you instead say "the called function," but "callee" is more common and pairs with "caller."

**Review by scottamain — CHANGES_REQUESTED** [review]
Thank you! Looks good, except the edit in the lifetimes doc is not quite right.

**Comment by Mayureshd-18** [comment]
@scottamain Changed the "callee". Please review.
Regards

**Review by scottamain — APPROVED** [review]
Thank you!

**Pull request merged**